### DISCUSSION TITLE
Output of transpiled async code not the same as untranspiled async code

### DISCUSSION THREAD

**Discussion by danieltroger** [parcel-bundler/parcel#6476] [Q&A] [answered]
@mischnic Can you tell me if code transpiled from async to sync done by babel/swc using regenerator runtime is meant to output the same things in all cases as the untanspiled code?

If so I found a bug.

But maybe it's just some known limitation? Where are there docs? https://github.com/facebook/regenerator/tree/master/packages/regenerator-runtime does not exist.

I kinda stretched promises and async to the extreme and have a very nice thing working but it outputs differently untranspiled and transpiled. Basically the transpiled code prioritises different pieces of code wrong and therefore doesn't produce the wanted output.

Also it's so complicated and hard to debug that I wonder if it's even worth trying

**Comment by mischnic** [comment, marked as answer]
The behaviour before transpiling should *always* be exactly the same as the behaviour after transpiling. Everything else is a bug (unless you told a transpiler to take shortcuts, like babel-preset-env's loose mode)

**Reply by danieltroger** [reply]
Nice! Can I privately send you my code and you take a look? 👀

**Reply by mischnic** [reply]
Sure

**Reply by danieltroger** [reply]
Amazing, sent you a dm on twitter!
